src/Collections/Workflows.php

```diff
@@ -0,0 +1,249 @@
+    {
+        $workflows = array_filter(
+            $this->all(),
+            function ($workflow) {
+                $is_finished = $workflow->isFinished();
+                return $is_finished;
+            }
+        );
+        return $workflows;
+    }
+
+    /**
+     * Returns all existing workflows that contain logs
+     *
+     * @return Workflow[]
+     */
+    public function allWithLogs()
+    {
+        $workflows = $this->allFinished();
+        $workflows = array_filter(
+            $workflows,
+            function ($workflow) {
+                $has_logs = $workflow->get('has_operation_log_output');
+                return $has_logs;
+            }
+        );
+
+        return $workflows;
+    }
+
+    /**
+     * Creates a new workflow and adds its data to the collection
+     *
+     * @param string $type Type of workflow to create
+     * @param array $options Additional information for the request, with the
+     *   following possible keys:
+     *   - environment: string
+     *   - params: associative array of parameters for the request
+     * @return Workflow $model
+     */
+    public function create($type, array $options = [])
+    {
+        $options = array_merge(['params' => [],], $options);
+        $params = array_merge($this->args, $options['params']);
+
+        $results = $this->request->request(
+            $this->url,
+            [
+                'method' => 'post',
+                'form_params' => [
+                    'type' => $type,
+                    'params' => (object)$params,
+                ],
+            ]
+        );
+
+        $model = new $this->collected_class(
+            $results['data'],
+            ['id' => $results['data']->id, 'collection' => $this,]
+        );
+        $this->add($model);
+        return $model;
+    }
+
+    /**
+     * Returns the object which controls this collection
+     *
+     * @return mixed
+     */
+    public function getOwnerObject()
+    {
+        if (isset($this->environment)) {
+            return $this->environment;
+        } elseif (isset($this->organization)) {
+            return $this->organization;
+        } elseif (isset($this->site)) {
+            return $this->site;
+        } elseif (isset($this->user)) {
+            return $this->user;
+        }
+        return null;
+    }
+
+    /**
+     * Fetches workflow data hydrated with operations
+     *
+     * @param array $options Additional information for the request
+     * @return void
+     */
+    public function fetchWithOperations($options = [])
+    {
+        $options = array_merge(
+            $options,
+            ['fetch_args' => ['query' => ['hydrate' => 'operations',],],]
+        );
+        $this->fetch($options);
+    }
+
+    /**
+     * Get most-recent workflow from existing collection that has logs
+     *
+     * @return Workflow|null
+     */
+    public function findLatestWithLogs()
+    {
+        $workflows = $this->allWithLogs();
+        usort(
```

Comment:
Yeah. There are lots of places to gain efficiency in some of these models. I tried to leave the business logic more or less alone for now though.